SalaryService.Api/Controllers/CompensationController.cs

```diff
@@ -28,15 +29,31 @@ public async Task CreateAsync([FromBody] CompensationCreateDto dto)
         await _compensationsService.CreateAsync(dto, employee);
     }
 
+    [RequiresPermission(UserClaimsProvider.ViewPersonalCompensations)]
     [HttpGet("all")]
     public async Task<CompensationListDto> GetAllAsync()
     {
-        return await _compensationsService.GetAllAsync();
+        return await _compensationsService.GetAllAsync(User.GetCorporateEmail());
     }
 
+    [RequiresPermission(UserClaimsProvider.ViewPersonalCompensations)]
     [HttpGet("types")]
     public async Task<List<CompensationType>> GetTypesAsync()
     {
         return await _compensationsService.GetTypesAsync();
     }
+
+    [RequiresPermission(UserClaimsProvider.ViewEmployeesCompensations)]
+    [HttpGet("admin/all")]
+    public async Task<CompensationCeoListDto> GetAdminAllAsync()
+    {
+        return await _compensationsService.GetAdminAllAsync();
+    }
+
+    [RequiresPermission(UserClaimsProvider.ViewEmployeesCompensations)]
+    [HttpPut("update")]
+    public async Task UpdateStatusAsync([FromQuery] string status, [FromBody] long[] compensationsIds)
+    {
+        await _compensationsService.UpdateStatusAsync((status == "paid") ? true : false, compensationsIds);
```

Comment:
I think it would be better to change the method signature. The first argument will be a string. Inside the method handle is status paid or not. Also `paid` should be a constant inside the service